sources/SilkTouch/SilkTouch/Mods/TransformCOM.cs

```diff
@@ -0,0 +1,749 @@
+            : CSharpSyntaxRewriter
+        {
+            public override SyntaxNode? VisitPointerType(PointerTypeSyntax node)
+            {
+                for (int i = 0; i < ComTypes.Count; i++)
+                {
+                    (string, bool, KeyedStringTree?) val = ComTypes[i];
+
+                    if (val.Item1 == node.ElementType.ToString())
+                    {
+                        return IdentifierName(val.Item1);
+                    }
+                }
+
+                return base.VisitPointerType(node);
+            }
+
+            public override SyntaxNode? VisitIdentifierName(IdentifierNameSyntax node)
+            {
+                if (node.Identifier.ToString() == "lpVtbl")
+                {
+                    return ParenthesizedExpression(PrefixUnaryExpression(SyntaxKind.PointerIndirectionExpression, node));
+                }
+
+                return base.VisitIdentifierName(node);
+            }
+
+            public override SyntaxNode? VisitVariableDeclaration(VariableDeclarationSyntax node)
+            {
+                if (node.Type.ToString() == "void**" && node.Variables.First().Identifier.ToString() == "lpVtbl")
+                {
+                    return VariableDeclaration(PointerType(PointerType(PointerType(IdentifierName("void")))))
+                        .AddVariables(VariableDeclarator("lpVtbl"));
+                }
+
+                return base.VisitVariableDeclaration(node);
+            }
+
+            public override SyntaxNode? VisitInterfaceDeclaration(InterfaceDeclarationSyntax node)
+            {
+                var ret = base.VisitInterfaceDeclaration(node);
+
+                if (ret is InterfaceDeclarationSyntax inter && inter.BaseList is not null && inter.BaseList.Types.Any(baseType => baseType.Type.ToString().StartsWith("I") && baseType.Type.ToString().EndsWith(".Interface")))
+                {
+                    List<BaseTypeSyntax> baseTypes = [];
+                    foreach (BaseTypeSyntax baseType in inter.BaseList.Types)
+                    {
+                        if (ComTypes.Any(com => $"{com.Item1}.Interface" == baseType.Type.ToString()))
+                        {
+                            baseTypes.Add(SimpleBaseType(IdentifierName(baseType.Type.ToString())));
+                        }
+                        else
+                        {
+                            baseTypes.Add(baseType);
+                        }
+                    }
+
+                    ret = inter.WithBaseList(BaseList(SeparatedList(baseTypes)));
+                }
+
+                return ret;
+            }
+
+            public override SyntaxNode? VisitCastExpression(CastExpressionSyntax node)
+            {
+                var castType = node.Type.ToString();
+
+                var expression = node.Expression.ToString();
+
+                if (expression != "Unsafe.AsPointer(ref this)")
```

Comment:
This is of course assuming that we're within the struct methods, which we might not be (unless I've missed something). Is this something to worry about?

Reply:
We may potentially need to handle the case where this happens outside of an internal struct function (which wouldn't live in the rewriter but rather the usageUpdater, but so far it hasn't been an issue so wasn't one of the cases I've covered there.

Reply:
This has been fixed with a few checks so it should only affect Interface types

Reply:
Ok, looks fine for now.